stdlib/public/RemoteInspection/TypeRefBuilder.cpp

```diff
@@ -538,8 +538,9 @@ bool TypeRefBuilder::getFieldTypeRefs(
     // We need this for enums; an enum case "is generic" if any generic type
     // parameter substitutions occurred on the payload.  E.g.,
     // `case a([T?])` is generic, but `case a([Int?])` is not.
-    bool IsGeneric = false;
-    auto Substituted = Unsubstituted->subst(*this, *Subs, IsGeneric);
+    bool IsGeneric = !Unsubstituted->isConcrete();
+    auto Substituted = (IsGeneric ? Unsubstituted->subst(*this, *Subs)
+                                  : Unsubstituted);
```

Comment:
@tbkka You added the DidSubstitute flag originally, so let me know if my change makes sense. I thought of adding an assertion here that `Unsubstituted` remains unchanged when `isConcrete()` returns true, which should be the case, but I know we don't like assertion failures in remote mirrors/lldb.

Reply:
Somewhere around here I have a couple of cases where `DidSubstitute` actually does the wrong thing.  I had dug into it enough to realize that it needs to be ripped out and replaced.

The remote mirror library gets invoked to parse damaged or incorrect metadata, so it has to fail cleanly in such cases.  That doesn't preclude using assertions, but they have to be limited to cases where we're validating the correct behavior of the _code_ and not the correctness of the data.